src/DataValues/DecimalValue.php

```diff
@@ -47,39 +47,31 @@ class DecimalValue extends DataValueObject {
 	 *
 	 * @param string|int|float $value If given as a string, the value must match
 	 *                         QUANTITY_VALUE_PATTERN.
+	 *
+	 * @throws IllegalValueException
 	 */
 	public function __construct( $value ) {
 		if ( is_int( $value ) || is_float( $value ) ) {
 			$value = $this->convertToDecimal( $value );
-		}
-
-		$this->assertNumberString( $value );
-
-		// make "negative" zero positive
-		$value = preg_replace( '/^-(0+(\.0+)?)$/', '+\1', $value );
-
-		$this->value = $value;
-	}
-
-	/**
-	 * Checks that the given value is a number string.
-	 *
-	 * @param string $number The value to check
-	 *
-	 * @throws IllegalValueException
-	 */
-	private function assertNumberString( $number ) {
-		if ( !is_string( $number ) ) {
+		} elseif ( !is_string( $value ) ) {
 			throw new IllegalValueException( '$number must be a numeric string.' );
 		}
 
-		if ( strlen( $number ) > 127 ) {
+		$value = trim( $value );
+
+		if ( strlen( $value ) > 127 ) {
 			throw new IllegalValueException( 'Value must be at most 127 characters long.' );
 		}
-
-		if ( !preg_match( self::QUANTITY_VALUE_PATTERN, $number ) ) {
+		if ( !preg_match( self::QUANTITY_VALUE_PATTERN, $value ) ) {
```

Comment:
True, but it's a constant that may be used elsewhere... since this patch is about properly trimming, and fixing the odd behavior of $ in pcre, it seems a good time and place to turn the $ into a \z